src/main/java/edu/springsecurity/hw15/service/UserServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package edu.springsecurity.hw15.service;
+
+import edu.springsecurity.hw15.model.Role;
+import edu.springsecurity.hw15.model.User;
+import edu.springsecurity.hw15.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService, UserDetailsService {
+
+    private final UserRepository userRepository;
+
+    @Override
+    public void saveUser(User user) {
+        Set<Role> roles = new HashSet<>();
+        roles.add(Role.USER);
+        userRepository.save(new User(user.getUsername(), user.getEmail(), user.getPassword(), roles));
+    }
+
+    @Override
+    public User updateUser(User user) {
+
+        return userRepository.saveAndFlush(user);
+    }
+
+    @Override
+    public List<User> findAllUsers() {
+
+        return userRepository.findAll();
+    }
+
+    @Override
+    public Optional<User> getUserById(Long id) {
```

Comment:
You should return User, not Optional just use orElseThrow(() -> new DataNotFoundException("User not found"))